src/login7-payload.js

```diff
@@ -392,7 +393,17 @@ class Login7Payload {
           buffer.writeUInt8(FEDAUTH_OPTIONS.FEATURE_ID, 0);
           buffer.writeUInt32LE(2, 1);
           buffer.writeUInt8((FEDAUTH_OPTIONS.LIBRARY_ADAL << 1) | (fedAuth.echo ? FEDAUTH_OPTIONS.FEDAUTH_YES_ECHO : FEDAUTH_OPTIONS.FEDAUTH_NO_ECHO), 5);
-          buffer.writeUInt8(fedAuth.workflow === 'integrated' ? 0x02 : FEDAUTH_OPTIONS.ADAL_WORKFLOW_USER_PASS, 6);
+
+          switch (fedAuth.workflow) {
+            case 'integrated':
+              buffer.writeUInt8(FEDAUTH_OPTIONS.ADAL_WORKFLOW_INTEGRATED, 6);
+              break;
+
+            case 'default':
+              buffer.writeUInt8(FEDAUTH_OPTIONS.ADAL_WORKFLOW_USER_PASS, 6);
+              break;
+          }
```

Comment:
It should throw the `Unrecognized Authentication type for fedauth ADAL request` by default, like in [SQLClient](https://referencesource.microsoft.com/#System.Data/fx/src/data/System/Data/SqlClient/TdsParser.cs,7029) in C#

```C#
switch (fedAuthFeatureData.authentication) {
    case SqlAuthenticationMethod.ActiveDirectoryPassword:
        workflow = TdsEnums.ADALWORKFLOW_ACTIVEDIRECTORYPASSWORD;
        break;
    case SqlAuthenticationMethod.ActiveDirectoryIntegrated:
        workflow = TdsEnums.ADALWORKFLOW_ACTIVEDIRECTORYINTEGRATED;
        break;
    case SqlAuthenticationMethod.ActiveDirectoryInteractive:
        workflow = TdsEnums.ADALWORKFLOW_ACTIVEDIRECTORYINTERACTIVE;
        break;
    default:
        Debug.Assert(false, "Unrecognized Authentication type for fedauth ADAL request");
        break;
}
```

Reply:
@funkydev The correct type is already enforced via the configuration. 😄